packages/cadl-ranch-specs/http/azure/core/page/main.tsp

```diff
@@ -200,3 +201,392 @@ interface TwoModelsAsPageItem {
+model TransactionRecord {
+  /**
+   * Contents of the ledger entry.
+   */
+  contents: string;
+
+  @visibility("read")
+  collectionId: string;
+
+  @key
+  @visibility("read")
+  transactionId: string;
+}
+
+/**
+ * Type for input model body
+ */
+model ListPaginationLedgerEntryInputBody {
+  /**
+   * A required string parameter
+   */
+  requiredString: string;
+
+  /**
+   * A required integer parameter
+   */
+  requiredInt: int32;
+}
+
+/**
+ * Input model for the ListPaginationLedgerEntries
+ */
+model ListPaginationLedgerEntryInput {
+  /**
+   * Body parameter for the ListPaginationLedgerEntries input model
+   */
+  @body bodyInput: ListPaginationLedgerEntryInputBody;
+}
+
+/**
+ * Paginated ledger entries
+ */
+@global.Azure.Core.pagedResult
+model PagedLedgerEntries {
+  /**
+   * Array of ledger entries
+   */
+  @global.Azure.Core.items
+  entries: TransactionRecord[];
+
+  /**
+   * Path to retrieve next page of ledger entries
+   */
+  @global.Azure.Core.nextLink
+  nextPageLink?: ResourceLocation<TransactionRecord>;
+}
+
+@scenario
+@scenarioDoc("""
+  Expected input body:
+  ```json
+  {
+    "requiredString": "string",
+    "requiredInt": 1
+  }
+  ```
+  
+  Expected response body:
+  ```json
+  {
+    "entries": [
+      {
+        "contents": "string",
+        "collectionId": "string",
+        "transactionId": "string"
+      }
+    ]
+  }
+  """)
+@route("/ledger-entries")
+op ListPaginationLedgerEntries is global.Azure.Core.Foundations.ResourceList<
+  TransactionRecord,
+  ListPaginationLedgerEntryInput,
+  PagedLedgerEntries
+>;
+
+/**
+ * The interval
+ */
+union Interval {
+  /**
+   * A string representing a custom interval.
+   */
+  string,
+
+  PT5S: "PT5S",
+  PT10S: "PT10S",
+  PT1M: "PT1M",
+  PT5M: "PT5M",
+  PT1H: "PT1H",
+}
+
+model MetricDimensions {
+  /**
+   * The interval (i.e. timegrain) of the query.
+   */
+  @query
+  interval?: Interval;
+
+  /**
+   * Metric name
+   */
+  @query
+  metricName?: string;
+
+  /**
+   * Metric namespace to query metric definitions for.
+   */
+  @query
+  metricNamespace: string;
+
+  /**
+   *The timespan of the query.
+   */
+  @query
+  timespan?: string;
+}
+
+@resource("values")
+@parentResource(Dimension)
+model DimensionValueListItem {
+  /**
+   * The values of the dimension.
+   */
+  @key
+  value: string[];
+}
+
+@resource("dimensions")
+@parentResource(TestRuns)
+model Dimension {
+  @key
+  @path("name")
+  @visibility("query")
+  name: string;
+}
+
+/**
+ * Test run model
+ */
+@resource("test-runs")
+model TestRuns {
+  /**
+   * Unique test run name as identifier
+   */
+  @key
+  @path
+  @visibility("read")
+  testRunId: string;
+}
+
+@scenario
+@scenarioDoc("""
+  List the dimension values for the given metric dimension name.
+  """)
+interface AdditionalParameter {
+  /**
+   * List the dimension values for the given metric dimension name.
+   */
+  @route("/metric-dimensions")
+  ListMetricDimensionValues is global.Azure.Core.ResourceList<
```

Comment:
can these names be more clear? it's important that the coverage scenario clearly tells what this is testing. `MetricDimensionValues` doesn't have much meaning. For each of these operations, can you name them so we can explicitly see what we're testing that we didn't test before?